objects/screen.c

```diff
@@ -117,14 +117,29 @@
  */
 
 /**
- * The screen number.
+ * The internal screen number.
  *
- * An integer greater than 1 and smaller than `screen.count()`. Please note that
- * the screen order isn't always mirroring the screen logical position.
+ * * The indeces are a continuous sequence from 1 to `screen.count()`.
+ * * It is **NOT** related to the actual screen position relative to each
+ *   other.
+ * * 1 is **NOT** necessarily the primary screen.
+ * * When screens are added and removed indices **CAN** change.
+ *
+ * If you really want to keep an array of screens you should use something
+ * along:
+ *
+ *     local myscreens = setmetatable({}. {__mode="k"})
+ *     myscreens[ screen[1] ] = "mydata"
```

Comment:
Why not just use `screen[1].mydata` directly instead? That has way less chance of causing problems with the garbage collector. (I guess this PR is so old already, that back then we did not yet have this feature?)

Reply:
> I guess this PR is so old already

Yes, ancient, this is a copy of object.ldoc with `/* */` instead of `--`. I will ajust both.

Before this PR, client:connect_signal was never in the documented API